<!--
%\VignetteEngine{knitr::docco_linear}
%\VignetteIndexEntry{fullsibQTL}
-->

# fullsibQTL Tutorial

```{r, globalsetup, echo=FALSE, results='hide', cache=FALSE}
#opts_chunk$set(cache=FALSE, autodep=TRUE)
knitr::opts_chunk$set(fig.width=10)
```

This tutorial was developed under the following softwares: `R 3.4.0`, `onemap 2.0.7`, and `fullsibQTL 1.0.0`.

`fullsibQTL` is an `R` package to perform QTL mapping in outbred/outcrossing species. We consider as a mapping population a full-sib progeny (or *F1* population) derived by a bi-parental cross between two non homozygous parents, with a genetic map obtained with markers showing different segregation patterns. Here, we assumed the phenotypes are continuous having normal distribution and the genetic map was previously obtained with `onemap` package. If you are not familiar with `onemap`, we strongly encourage you to read `onemap` [vignettes](https://github.com/augusto-garcia/onemap/tree/master) (Margarido et al., 2007).

In our software, we developed tools to perform Composite Interval Mapping (CIP). Briefly, we developed an statistical model with three genetics effects, one for each parent and an interaction (dominance). To obtain these estimates, we used maximum likelihood approach using mixture models and EM algorithm (Gazaffi et al., 2014). The QTL genotype probabilities were calculated using a multipoint technology, based on Hiddden Markov Models (Wu et al., 2002b). We also implemented a function to select cofactors using multiple linear regression and the information criteria. Permutation test (Churchill and Doerge, 1994) for threshold determination were also implemented, as well the modification proposed by Chen and Storey (2006) for having a relaxed threshold. Some extra functions were developed to provide a graphical or text output for the analysis helping the interpretation of QTL mapping. To exemplify the usage of `fullsibQTL`, here we analyse the dataset present by Gazaffi et al. (2014).

The purpose of this tutorial is to help users dealing with this package and understanding the outputs. It is not our intention to teach the genetic basis of QTL mapping approach, for a complete understand about the method see Gazaffi et al. (2014). Remembering, users of `fullsibQTL` are supposed to have some experience with R, since the analysis is done using the command line. Previous knowledge of `onemap` is also desirably, once the genetic map is obtained with this software. `fullsibQTL` is available as source code for Windows and Linux. It is released under the GNU General Public License, is open-source and the code can be changed freely. It comes with no warranty. It is implemented as a package to be used under the freely distributed R software, which is a language and environment for statistical computing

If you are not familiar with `R`, we recommend the reading of
vignette [Introduction to R](http://htmlpreview.github.io/?https://github.com/augusto-garcia/onemap/blob/master/inst/doc/Introduction_R.html). If you are not familiar with `onemap`, we recommend the reading of vignette [How to build a linkage map for outcrossing populations](http://htmlpreview.github.io/?https://github.com/augusto-garcia/onemap/blob/master/inst/doc/Outcrossing_Populations.html).

Additionally, this tutorial has three appendices:

- [Appendix A: QTL mapping with partially informative markers](#appendixa)

- [Appendix B: Graphical options with `plot_fullsibQTL`](#appendixb)

- [Appendix C: Graphical options with `plot`](#appendixc)

## Citation
To cite the methodology for QTL mapping in full-sib progeny:
Gazaffi, R., Margarido, G. R., Pastina, M. M., Mollinari, M., Garcia, A. A. F. (2014). A model for quantitative trait loci mapping, linkage phase, and segregation pattern estimation for a full-sib progeny. *Tree Genetics & Genomes*, 10(4), 791-801.

To cite this R package:
Gazaffi, R., Amadeu, R. R., Rosa, J. R. B. F., Garcia, A. A. F. (2017). Application notes: fullsibQTL: R package for QTL mapping using a full-sib progeny, *Journal*, v(x), pages().

### Installation from CRAN (stable version)
Within R:
```{r, eval=FALSE}
install.packages("fullsibQTL")
```

### Installation from github (version under development)
Within R:
```{r, eval=FALSE}
## Installing and loading devtools
install.packages("devtools")
library(devtools)

## Installing from github the fullsibQTL
install_github("augusto-garcia/fullsibQTL")
```

## Functions
The following table presents the `fullsibQTL` functions which can be directly called by users:

<font size="2">

| Type                       | Name                       | Description                                                                                    |
|----------------------------|----------------------------|------------------------------------------------------------------------------------------------|
| input                      | `read_outcross_pheno`      | Reads the data file containing markers and phenotypic values                                   |
|                            | `create_fullsib`           | Creates the object to perform QTL                                                              |
| &nbsp; | | |
| interval mapping           | `im_scan`                  | Scan the genome using IM approach                                                              |
|                            | `im_char`                  | Provides the genetic effects, LOD Score and the p-values of complementary testes               |
| &nbsp; | | |
| cofactors                  | `cof_selection`            | Selects cofactors using multiple linear regression                                             |
|                            | `cof_definition`           | Defines locations on the genome to be used as cofactors                                        |
| &nbsp; | | |
| composite interval mapping | `cim_scan`                 | Scan the genome using CIM approach                                                        |
|                            | `cim_char`                 | Provides the genetic effects, LOD Score and the p-values of complementary testes               |
| &nbsp; | | |
| summaries <br> (S3 Methods)  | `print`                   | class `fullsib`: Prints a summary for the object of class fullsib                              |
| 			     | `print`                    | class `fullsib_scan`: Prints the result of im_scan or cim_scan                                 |
|                            | `summary`                  | class `fullsib_scan`: Summarizes the QTL search for im_scan or cim_scan                        |
|                            | `plot`                     | class `fullsib_scan`: Plot the QTL profile for the mapped groups                               |
|                            | `summary`                  | class `fullsib_perm`: Provides the threshold values for permutations test                      |
|                            | `plot`   | class `fullsib_perm`: Plot the empiral distribution for permutations test					   |
|                            | `draw_phase`               | Returns the linkage phase between QTL and markers                                              |
|                            | `get_segr`                 | Returns the QTL segregation                                                                    |
|                            | `r2_ls`                    | Provides the phenotypic proportion of each QTL mapped and altogether (least square estimation) |

</font>

## Required data
The input file of `fullsibQTL` is the same as the one required by `onemap` package, you can find more information at `onemap` [vignette](http://htmlpreview.github.io/?https://github.com/augusto-garcia/onemap/blob/master/inst/doc/Outcrossing_Populations.html). Above there is an example of such file. The header line indicates the data type, the following line with `10 5 0 0 2` indicates that the data has, respectively, 10 individuals, 5 markers, no (0) chromosome information, 0 (no) physical position information, and 2 traits as phenotypic data. It is very similar to a `MAPMAKER/EXP` file, but has additional information about the crosstype. It uses the symbol `-` for missing data.

```
data type outcross
10 5 0 0 2
I1 I2 I3 I4 I5 I6 I7 I8 I9 I10
*M1 B3.7 ab ab - ab b ab ab - ab b
*M2 D2.18 o - a a - o a - o o
*M3 D1.13 o a a o o - a o a o
*M4 A.4 ab b - ab a b ab b - a
*M5 D2.18 a a o - o o a o o o
*pheno1 4.8 2.1 10.6 3.7 -3.7 -7.5 - 3.9 3.6 -5.8
*pheno2 5 -5.8 - 14.8 -2.4 -1.9 -1.1 8.1 -11.1 4.8
```

## Creating the working object
Hopefully you already had used the `onemap` to build your linkage maps an also are familiar with this data type. Within the available file, you can import it. First you need to read it with `onemap::read_onemap` function and after you can combine it with your linkages groups with `create_fullsib` function. Your final object will have the `onemap` raw object, the linkage groups and some parameters information. In the following example we import the raw data and a `list` with 4 linkages groups, we set as mapping function `kosambi` (Kosambi, 1944) with `map.function` argument, and, at last, we set the `step` argument as 1. The `step` argument is used for the computation of the conditional multipoint probability which defines the distance in cM (centiMorgan) the conditional probability are obtained. If `step=0`, the probabilities will be computed only in the positions of each marker.

```{r, eval=FALSE}
library(fullsibQTL)
fs_data <- read_onemap(dir = "C:/workingdirectory", inputfile = "filename.raw")

## To read the example data from this same package:
fs_data <- read_onemap(dir = system.file("extdata", package = "fullsibQTL"), 
                       inputfile = "example_QTLfullsib.raw")

## Within the Linkage Groups done:
fsib <- create_fullsib(fs_data,
                       map.list = list(LG1_final, LG2_final, LG3_final, LG4_final),
                       step = 1, map.function = "kosambi")
```

After loading the data, the user can look at the the object's summary with:
```{r, eval=FALSE}
fs_data
fsib
```

If you want to see this data file in its raw format, type the following line to see where it is located:
```{r, eval=FALSE}
system.file("extdata", "example_QTLfullsib.raw", package = "fullsibQTL")
```

```{r, eval=TRUE, echo=FALSE}
rm(list=ls())
library(fullsibQTL)
```

In this tutorial we will used the data example available in the package. In this data, there is 6 objects, a raw data in the `onemap` format (`example_QTLfullsib`), 4 ordered linkage groups (`LG1_final`, `LG2_final`, `LG3_final`, and `LG4_final`), and 2 objects with permutations which will be later explained (`im_perm` and `cim_perm`). To load it:
```{r, eval=TRUE,cache=FALSE}
data(example_QTLfullsib) ## Loading the example data
ls() ## Checking if the objects were loaded
example_QTLfullsib
LG1_final
```

Now we have to combine the raw data with the linkage groups (hereafter LG) in a single object with the function `create_fullsib`:
```{r, eval=TRUE, cache=FALSE}
fsib <- create_fullsib(example_QTLfullsib,
                       map.list=list(LG1_final, LG2_final, LG3_final, LG4_final),
                       step=1, map.function="kosambi")
fsib
```

This object contains the genetic map composed by 4 LGs, with their length and segregation patterns above indicated. There is also shown the number of unlinked makers, *i.e.*, how many markers of the `example_QTLfullsib` object are not present in any LG. This may be helpful for species where the map does not cover the whole genome. Finally, there is the number of the available phenotypes for the QTL mapping, as well the chosen distance (`step`) to obtain the conditional probability and the mapping function.  It is also keep the map function, in this case `kosambi`. In this tutorial we will use just the first trait.

## Interval mapping
In order to proceed with interval mapping, there are two implemented functions `im_scan` and `im_char`. `im_scan` scans all the genome computing associations between genotype and phenotype trying to detect QTLs. `im_char` shows the genetic effects and their significance tests. It also shows additional tests to infer the QTL segregation pattern.

### Genome scan
The genome scan can be performed with the function `im_scan`, such as:
```{r, eval=TRUE, cache=FALSE}
im <- im_scan(fullsib = fsib, lg="all", pheno.col=1, LOD=TRUE)
```
The argument `fullsib` refers to the object created using `create_fullsib`, `lg` indicates which linkage groups must be scanned for the QTL mapping. Another way to represent it is `lg=1:4`. The function default is `lg=all`, *i.e.*, scan every LG. The argument `pheno.col` indicates which phenotype must be used, default is to use the first trait, it must be an integer. The argument `LOD=TRUE` indicates that the mapping result must be shown with the LOD score, if `FALSE`, the scale will be shown as $-log_{10}(p_{value})$.

The mapping result can be view calling (here is truncated):
```{r, eval=TRUE, cache=FALSE, R.options=list(max.print=50)}
im
```

Another way to see the results is using `print` and specifying which LG you would like to print:
```{r, eval=TRUE, cache=FALSE, R.options=list(max.print=50)}
print(im, lg=1)
```
The resulting object of `im_scan` is a matrix with 4 columns and *k* lines, where *k* is the number of genome positions in which the conditional probabilities have been computed based on the `step` argument of the earlier `create_fullsib` function. In this object, the row names are the identification of each position (it can be the marker name itself or a pseudo-marker with a automatically given name in the format `loc` + number. The first column is the linkage group of the marker. The third column is the position position (in centiMorgan). The fourth column is the LOD score value (or the $-log_{10}(p_{value})$). The fifth column reefers to the model utilized for the computations (see Section 9).

`summary(im)` prints just the utmost LOD score value (or the $-log_{10}(p_{value})$) for every LG. The user can also set a threshold (`thr` argument) where just the scores above this value will be printed. When more than one QTL is mapped in the same LG, the user have to do manual QTL search as looking for more than one peak can be a subjective process.

```{r, eval=TRUE, cache=FALSE}
summary(im)
```

#### QTL mapping plot options
To obtain a graphical plot about the QTL mapping you have two options: `plot` and `plot_fullsibQTL`. `plot` is the lighter one and should be used for faster purposes. `plot_fullsibQTL` is the heavier one and should be used for analytical purposes.  

For `plot`:
```{r, eval=TRUE, cache=FALSE}
plot(im)
```

The user can customize the QTL graphic plot. As an example, it is possible to plot in the same area until 5 mapping profiles. Moreover, the user can indicate the molecular markers' position, change the labels, between others features. You can find more information about graphics at [plot Graphical options](#appendix:-graphical-options-with-`plot`) section and at the function help `?plot.fullsib_scan`.

Another user option is to use the `plot_fullsibQTL` function which is a more informative graphic (and also would result in a heavier graphic):

```{r, eval=TRUE, cache=TRUE}
plot_fullsibQTL(fullsib=fsib, fullsib.scan=im)
```

Within such function there the option to be a interactive graphic:
```{r, eval=FALSE, cache=FALSE}
plot_fullsibQTL(fullsib=fsib, fullsib.scan=im,interact = TRUE)
```

<iframe src='http://statgen.esalq.usp.br/fullsibQTL_graphics/im.html', height="600", width = "800">

</iframe>

You can find more information about `plot_fullsibQTL` in [Appendix B](#appendixb)

You can find more information about `plot` in [Appendix C](#appendixc)

In order to have a lighter tutorial, here, we will use just the graphics from `plot` function. However, we strongly recommend to use the `plot_fullsibQTL` graphics in your analysis.

### Covariate
Finally, to improve the mapping flexibility, it is possible to set a covariate (as a fixed effect) within the model in order to control the variability of the aimed phenotype. To set a full-rank matrix which controls the phenotypic variance, first you need to define a incidence vector, and, then, set it as an argument of the `im_scan`. An example follows:

```{r, eval=TRUE, cache=FALSE}
covar <- matrix(rep(c(1,-1), each=150), ncol=1)
im_covar <- im_scan(fsib, pheno.col=1, addcovar=covar)
```

### Permutation test
The threshold for QTL detection can be set using permutation tests (Churchill and Doerge, 1994) adding the argument `n.perm` in the `im_scan` function. As the permutation test is a re-sample process, it is recommended to the user to set a seed before the permutations. With this seed, the permutation can be repeated achieving the same result. The argument `write.perm` allows the user to save a text file with all the permutations performed. In this tutorial, the permutation test object `im_perm` with the seed `1234` is already created and is part of the `example_QTLfullsib` ealier loaded, there is no necessity to run the following lines. However, an example of how to perform the permutation follows:
```{r, eval=FALSE, cache=FALSE, collapse=TRUE}
## Perfoming the permutation
set.seed(1234)
im_perm <- im_scan(fsib, pheno.col=1, n.perm=1000, write.perm="~/Documents/im_permutations.txt")
```

If the user wants to keep the resulting permutation object to later analysis, it can be saved and load as an `Rdata` object:
```{r, eval=FALSE, cache=FALSE, collapse=TRUE}
save(im_perm,file="~/Documents/im_permutations.Rdata")
load("~/Documents/im_permutations.Rdata")
```

The permutation test demands some time. At the end of the process, `im_perm` receives a matrix with *p* lines, where *p* is the number of the realized permutations, and two columns with the highest and second highest peak values. The definition of the second peak is the same of the method two of Chen and Storey (2006), *i.e*, the highest LOD score (or $-log_{10}(p_{value})$) of the genome disregarding the LG where the highest peak was achieved. This threshold values can be visualized through `summary` at the object with the permutation tests. In this case, the user can set an $\alpha$ significance threshold. The result will be the quantile $(1-\alpha) \times 100$ of the samples. The function default is to show $\alpha$ values equal to $0.10$ and $0.05$.
```{r, eval=TRUE, cache=FALSE}
## Summary of the permutation test
summary(im_perm)
summary(im_perm, alpha=0.05)

## Getting the peaks for alpha 0.05. 
## between () for created and print the object
(im.peak1 <- summary(im_perm,alpha=0.05)[1,1])
(im.peak2 <- summary(im_perm,alpha=0.05)[1,2])

## Ploting the thrshold with the second peak
plot(im)
abline(h=im.peak2, col="red")

## Plotting both thresholds
plot(im, lty = 1, lwd = 2, incl.mkr = NULL, cex.incl = 0.7, 
       cex.axis = 0.8, col = "blue", ylab = "LOD Score", xlab = "Linkage Group", 
       main = "IM - Trait 1")
abline(h = im.peak1, lty = 2, lwd = 1, col = "black")
abline(h = im.peak2, lty = 2, lwd = 2, col = "black")
```


If the user wants to see the maximum distribution, he or she can use the `plot` function with `perm` object. If the user set the `peak` argument for `2`, the distribution of the second highest peaks will be shown. The vertical line represents the cut-off threshold for $\alpha=0.05$.

```{r, eval=TRUE, cache=FALSE}
plot(im_perm, peak=1)
abline(v=im.peak1)
abline(v=im.peak1,col="red")

plot(im_perm, peak=2)
abline(v=3.419819,col="blue")
abline(v=2.497431,col="red")
```

### QTL characterization
After the determination of the number and position of the QTLs, the user can characterize the QTLs. In other words, to identify the significant effects, the segregation pattern and linkage phases between QTLs and flanking markers. The already created object `im` shows the presence of at least 4 QTLs:

```{r, eval=TRUE, cache=FALSE}
summary(im)
```

The characterization of these 4 regions is realized with the function `im_char`:
```{r, eval=TRUE, cache=FALSE}
qtl1 <- im_char(fsib, pheno.col=1, lg=1, pos="loc4")
qtl2 <- im_char(fsib, pheno.col=1, lg=2, pos="M27")
qtl3 <- im_char(fsib, pheno.col=1, lg=3, pos="loc64")
qtl4 <- im_char(fsib, pheno.col=1, lg=4, pos="loc56")
```

It is worth noting that the positions "loc4", "M27", "loc64", and "loc56" are the ones presented in the `im` object kept when the `summary` function was applied. The object with the LG 1 QTL characterization  is:
```{r, eval=TRUE, cache=FALSE}
qtl1
```
`qtl1` is a matrix with 1 column and 15 lines. The first line has the LG information. The second line has the position in centiMorgan which the region is localized. The third and fourth lines shows, respectively, $-log_{10}(p_{value})$) and LOD Score, values next to the ones obtained using `im_scan`. The fifth line (`mu`) is the intercept effect of the model. The lines `alpha.p`, `alpha.q`, and `delta.pq` represent the QTL genetic effects. The lines `LOD.alpha.p`, `LOD.alpha.q`, and `LOD.delta.pq` are the statistics of the marginal tests of these genetic effects. All of these three tests was performed with 1 degree of freedom (testing just if the effect differs than zero). In this case, a LOD score above or equal $0.83$ indicates a significant given effect (LOD $= 0.83$ is the same as a LRT $= 3.81$ which under $\chi^2$ distribution with 1 degree of freedom has significance level of $5%$). The lines `H4.pvalue`, `H5.pvalue`, and `H6.pvalue` show the p-values for the complementary test in order to detect the QTL segregation. For more information about it, please see Gazaffi et al. (2014). At last, the line `model` shows which model was studied, see details on section 9. In some cases, a given test cannot be evaluated, in these cases the cells will have `NA` values.

We can print all the QTLs information with:
```{r, eval=TRUE, cache=FALSE}
cbind(qtl1,qtl2,qtl3,qtl4)
```
The identification of the linkage phases allows to detect where the allele which increase or reduce the effect are. Therefore, it is necessary an analysis of the `LOD.alpha.p` and `LOD.alpha.q` values. In order to facilitate interpretation, the function `draw_phase` was developed and can be used as follows:

```{r, eval=TRUE, cache=FALSE}
draw_phase(fsib, qtl1, 0.05)
```
The argument $\alpha$ equals $0.05$ is used to show the significant additive genetic effects. If it is the case, the allele with superscript 1 enhances the phenotype, and the allele with superscript 2 reduces it. Alleles with superscript 0 has no significant effect under this significance level threshold. To infer the QTL segregation, the user must use the `get_segr` function. Its default considers $\alpha=0.05$ of significance level for the marginal effects ("LOD.alpha.p","LOD.alpha.q","LOD.delta.pq") and the complementary tests ("H4.pvalue","H5.pvalue", "H6.pvalue"). It follows an example:

```{r, eval=TRUE, cache=FALSE}
get_segr(qtl1)
get_segr(qtl2)
get_segr(qtl3)
get_segr(qtl4)
```
## Cofactor selection
Before composite interval mapping (CIP), it is necessary to define which markers will be cofactors. To do it, there are two functions: `cof_selection` and `cof_definition`.

### Cofactor selection using multiple linear regression
The function `cof_selection` allows to select cofactors using multiple linear regression through *stepwise* approach. The criteria information is utilized to determine if a given possible cofactor is kept or not kept. Basically, this function first prepares the data and then uses `step` from `stats` package. For more information about `?cof_selection`. An example follows:

```{r, eval=TRUE, cache=FALSE}
cofs_fs <- cof_selection(fsib, pheno.col=1, k=log(300), n.cofactor = 10)
plot(cofs_fs)
```

The object `fsib` is the same one created using `create_fullsib`, `pheno.col` indicates which phenotypic is to be used in the analysis, `k` is the chosen penalty for the information criteria computation. If `k=2`, it will be used the Akaike Information Criteria. If `k=log(n)` where `n`is the mapping population size, it corresponds to the Bayesian Information Criteria. If the user wants another penalty values, we strongly recommend to look at the [QTL Cartographer v1.17 manual](http://statgen.ncsu.edu/qtlcart/manual.pdf), page 76 and the help of the R function `step`. For some situations, the process would recommend to keep several cofactors which can lead to overparametrization of the model. To control it, the argument `n.cofactor` allows to the user to restrict the number of selected cofactors of the model, *i.e*, the proceeding to keep cofactors is frozen when the number of kept cofactors is equal to `n.cofactor` (the dafult is `n.cofactor=10`.

The following example shows how to proceed the cofactor selection in a situation where the fixed covariates are set to control the character variability. To do it, we used the argument `addcovar`. In a similar way of `im_scan` function:

```{r, eval=TRUE, cache=FALSE}
covar <- matrix(rep(c(1,-1), each=150), ncol=1)
cofs_fs2 <- cof_selection(fsib, pheno.col=1, addcovar=covar, k=2, thres.effect=0.05)
```

After the cofactor selection, `thres.effect=0.05` verifies if there is no significant effects (with p-values higher than $0.05$). If true, the non-significant effects are removed keeping just the significant ones as cofactors. This is helpful to reduce the model's degree of freedom. It is worth noting that each cofactor can has until 3 genetics effects and not all three may be significant. The default of this argument is 1, *i.e*, there is no removal of non-significant markers. We strongly recommend that if you utilized this argument, you have to used by a given criteria not by the software's default.

If you want more information about this function, we recommend to look at the function help (`?cof_selection`).

### Cofactor ad-hoc definition
The user can also define the cofactors *ad-hoc* with the `cof_definition` function:
```{r}
cofs <- matrix(c("1","M2",
                 "2","M27",
                 "3","M33",
                 "4","M52"),4,2,byrow=TRUE)
cofs_fsdef <- cof_definition(fsib,pheno.col = 1,thres.effect = 0.05,cof.pos = cofs)
```

## Composite Interval mapping
In order to proceed with composite interval mapping, there are two implemented functions `cim_scan` and `cim_char`. They have analogous opitions to the `im_scan` and `im_char`. The first one scan all the genome computing associations between genotype and phenotype trying to detect QTLs. The second one shows the genetic effects and their significance tests. It also shows additional tests to infer the QTL segregation pattern. The mapping procedure is performed similar as present in im_scan. However the inclusion of cofactors can locate QTL with more precision with higher statistical power in comparison of interval mapping approach. For more information about `?cim_scan` and `?cim_char`.

### Genome scan
The genome scan can be performed with the function `cim_scan` indicating the cofactors object (class: `fullsib_cofactors`) ealier created with `cof_selection` or `cof_definition`.
```{r, eval=TRUE, cache=FALSE}
cim <- cim_scan(fullsib = cofs_fs, lg="all", ws=10, pheno.col=1, LOD=TRUE)
```

### Permutation test
The threshold for QTL detection can be set using permutation tests (Churchill and Doerge, 1994) adding the argument `n.perm` in the `cim_scan` function. As the permutation test is a re-sample process, it is recommended to the user to set a seed before the permutations. With this seed, the permutation can be repeated achieving the same result. The argument `write.perm` allows the user to save a text file with all the permutations performed. In this tutorial, the permutation test object `cim_perm` with the seed `4321` is already created and is part of the `example_QTLfullsib` ealier loaded, there is no necessity to run the following lines. However, an example of how to perform the permutation follows:
```{r, eval=FALSE, cache=FALSE}
set.seed(4321)
cim_perm <- cim_scan(fullsib = cofs_fs, lg="all", pheno.col=1, LOD=TRUE, n.perm=1000, write.perm="~/Documents/cim_permutations.txt")
```
If the user wants to keep the resulting permutation object to later analysis, it can be saved and load as an `Rdata` object:
```{r, eval=FALSE, cache=FALSE, collapse=TRUE}
save(cim_perm,file="~/Documents/cim_permutations.Rdata")
load("~/Documents/cim_permutations.Rdata")
```
The permutation test demands some time. At the end of the process, `cim_perm` receives a matrix with *p* lines, where *p* is the number of the realized permutations, and 4 columns representing the first, second, third, and fourth highest peak value respectively. The definition of the second peak is the same of the method two of Chen and Storey (2006), *i.e*, the highest LOD score (or $-log_{10}(p_{value})$) of the genome disregarding the LG where the highest peak was achieved. The third and fourth peak definitions are analogous to the second peak. This threshold values can be visualized through `summary` at the object with the permutation tests. In this case, the user can set an $\alpha$ significance threshold. The result will be the quantile $(1-\alpha) \times 100$ of the samples. The function default is to show $\alpha$ values equal to $0.10$ and $0.05$.
```{r, eval=TRUE, cache=FALSE}
## Summary of the permutation test
summary(cim_perm)
summary(cim_perm, alpha=0.05)

## Getting the peaks for alpha 0.05. 
## between () for created and print the object
(cim.peak1 <- summary(cim_perm, alpha=0.05)[1,1])
(cim.peak2 <- summary(cim_perm, alpha=0.05)[1,2])

## Ploting the thrshold with the second peak
plot(cim)
abline(h=cim.peak2, col="red")

## Plotting both thresholds
plot(cim, lty = 1, lwd = 2, incl.mkr = NULL, cex.incl = 0.7, 
     cex.axis = 0.8, col = "red", ylab = "LOD Score", xlab = "Linkage Group", 
     main = "CIM - Trait 1 - Rubber Tree")
abline(h = cim.peak1, lty = 2, lwd = 1, col = "black")
abline(h = cim.peak2, lty = 2, lwd = 2, col = "black")
```

If the user wants to see the maximum distribution, he or she can use the `plot` function with `perm` object. If the user set the `peak` argument for `2`, the distribution of the second highest peaks will be shown. The vertical line represents the cut-off threshold for $\alpha=0.05$.

```{r, eval=TRUE, cache=FALSE}
plot(cim_perm, peak=1)
abline(v=cim.peak1,col="red")

plot(cim_perm, peak=2)
abline(v=cim.peak2,col="red")
```

### QTL characterization
After the determination of the number and position of the QTLs, the user can characterize the QTLs. In other words, to identify the significant effects, the segregation pattern and linkage phases between QTLs and flanking markers. 

The already created object `cim` shows the presence of at least 4 QTLs:
```{r, eval=TRUE, cache=FALSE}
summary(cim)
```

If the following graphic shows more than those 4 QTL evidence, the user can map the others QTLs. With this graphic and the `cim_scan` result, it can be mapped the exact location of each QTL with resolution given by the ealier `step` argument. To do it, compare the LOD score in the `cim` object with the graphic interval. Although there are some packages and functions to find local maxima, we do not recommend to blindly use them. Those methods can return more QTLs than the reality, commonly there is report of inflection and false local maxima points.
```{r, eval=TRUE, cache=FALSE}
## Plotting the CIM results
plot(cim, lty = 1, lwd = 2, incl.mkr = NULL, cex.incl = 0.7, 
     cex.axis = 0.8, col = "red", ylab = "LOD Score", xlab = "Linkage Group", 
     main = "CIM - Trait 1 - Rubber Tree")
abline(h = cim.peak1, lty = 2, lwd = 1, col = "black")
abline(h = cim.peak2, lty = 2, lwd = 2, col = "black")

## Visual Analysis LG1
## QTLs peak at loc13 and M12
print(cim,lg = 1)
```

You can visualize the others LGs with likely apporach:

```{r, eval=FALSE, cache=FALSE}
## Visual Analysis LG2
## QTLs peak loc45, loc77, and loc 108
print(cim,lg = 2)

## Visual Analysis LG3
## QTLs peak at M33 and loc65
print(cim,lg = 3)

## Visual Analysis LG4
## QTL peak at M52
print(cim,lg = 4)
```

Through the visual analysis we found the following 8 QTLs peaks corresponding to locations next to the 8 cofactors.
```{r}
print(cim,lg=1,pos=c("loc13","M12","loc45"))
```

You can visualize the others QTLs with likely apporach:

```{r, eval=FALSE, cache=FALSE}
print(cim,lg=2,pos=c("loc45","loc77","loc108"))
print(cim,lg=3,pos=c("M33","loc65"))
print(cim,lg=4,pos="M52")
```

In order to estimate the total of proportion explained by the QTL and its characterization we can use the `cim_char` function:

```{r}
cim_QTL1_lg1 <- cim_char(fullsib = cofs_fs, lg=1, pheno.col=1,pos="loc13")
cim_QTL2_lg1 <- cim_char(fullsib = cofs_fs, lg=1, pheno.col=1,pos="M12")
cim_QTL1_lg2 <- cim_char(fullsib = cofs_fs, lg=2, pheno.col=1,pos="loc45")
cim_QTL2_lg2 <- cim_char(fullsib = cofs_fs, lg=2, pheno.col=1,pos="loc77")
cim_QTL3_lg2 <- cim_char(fullsib = cofs_fs, lg=2, pheno.col=1,pos="loc108")
cim_QTL1_lg3 <- cim_char(fullsib = cofs_fs, lg=3, pheno.col=1,pos="M33")
cim_QTL2_lg3 <- cim_char(fullsib = cofs_fs, lg=3, pheno.col=1,pos="loc65")
cim_QTL1_lg4 <- cim_char(fullsib = cofs_fs, lg=4, pheno.col=1,pos="M52")

## Printing the object
cim_QTL1_lg1
```

`cim_QTL1_lg1` is a matrix with 1 column and 15 lines. The flaking markers of this QTL are in the header. Then, the first line has the LG information. The second line has the position in centiMorgan which the region is localized. The third and fourth lines shows, respectively, $-log_{10}(p_{value})$) and LOD Score, values next to the ones obtained using `im_scan`. The fifth line (`mu`) is the intercept effect of the model. The lines `alpha.p`, `alpha.q`, and `delta.pq` represent the QTL genetic effects. The lines `LOD.alpha.p`, `LOD.alpha.q`, and `LOD.delta.pq` are the statistics of the marginal tests of these genetic effects. All of these three tests was performed with 1 degree of freedom (testing just if the effect differs than zero). In this case, a LOD score above or equal $0.83$ indicates a significant given effect (LOD $= 0.83$ is the same as a LRT $= 3.81$ which under $\chi^2$ distribution with 1 degree of freedom has significance level of $5%$). The lines `H4.pvalue`, `H5.pvalue`, and `H6.pvalue` show the p-values for the complementary test in order to detect the QTL segregation. For more information about it, please see Gazaffi et al. (2014). At last, the line `model` shows which model was studied, see details on section 9. In some cases, a given test cannot be evaluated, in these cases the cells will have `NA` values.

We can also print all the QTLs in a single table with:
```{r}
## Printing in a single table
cbind(cim_QTL1_lg1, cim_QTL2_lg1, cim_QTL1_lg2, cim_QTL2_lg2, cim_QTL3_lg2, cim_QTL1_lg3, cim_QTL2_lg3, cim_QTL1_lg4)
```

In order to estimate the QTL phases, we can use the `get_segr` function:
```{r}
get_segr(cim_QTL1_lg1, probs1= 0.05, probs2 = 0.05)
get_segr(cim_QTL2_lg1, probs1= 0.05, probs2 = 0.05)
get_segr(cim_QTL1_lg2, probs1= 0.05, probs2 = 0.05)
get_segr(cim_QTL2_lg2, probs1= 0.05, probs2 = 0.05)
get_segr(cim_QTL3_lg2, probs1= 0.05, probs2 = 0.05)
get_segr(cim_QTL1_lg3, probs1= 0.05, probs2 = 0.05)
get_segr(cim_QTL2_lg3, probs1= 0.05, probs2 = 0.05)
get_segr(cim_QTL1_lg4, probs1= 0.05, probs2 = 0.05)
```

We can draw the genetic map with the QTLs with its infered phases with:
```{r}
draw_phase(cofs_fs,cim_QTL1_lg1)
draw_phase(cofs_fs,cim_QTL2_lg1)
draw_phase(cofs_fs,cim_QTL1_lg2)
draw_phase(cofs_fs,cim_QTL2_lg2)
draw_phase(cofs_fs,cim_QTL3_lg2)
draw_phase(cofs_fs,cim_QTL1_lg3)
draw_phase(cofs_fs,cim_QTL2_lg3)
draw_phase(cofs_fs,cim_QTL1_lg4)
```

### Estimating the QTL effects with Least Squares
The `r2_l2` function calculates the proportion of all QTL together (and also separetely) explain from total phenotipic variation using least square estimation. For more information about `?r2_l2`. Here we proceed with the CIM analysis estimating its QTL effects:

```{r}
## Estimating QTL effects without covariate
r2_ls(fsib, pheno.col=1, lg=c(1,1,2,2,2,3,3,4),
     pos=c("loc13","M12","loc45","loc77","loc108","M33","loc65","M52"))
```
It returns a vector with $q+1$ position ($q$ is the number of QTL) with $R^2$ estimation. In the first position one finds the value for a model fitted with all QTLs simultaneously (identified as `R2.trait`), the others positions provides the $R^2$ for each QTL separately (identified as `R2.lgX.Y`, where `lgX.Y` means the LG number $X$ in the $Y$-th position)

There is also the possibility to estimate the QTL effects considering covariates as follows:
```{r}
## Estimating QTL effects with covariates
covar <- matrix(rep(c(1,-1), each=150), ncol=1)
r2_ls(fsib, pheno.col=2, addcovar = covar, lg=c(1,1,2,2,2,3,3,4),
      pos=c("loc13","M12","loc45","loc77","loc108","M33","loc65","M52"))
```
<a name="appendixa">
## Appendix A: QTL mapping with partially informative markers
</a>
The appendix must be read as Supplementary information of Gazaffi et al. (2014).

The model presented by Gazaffi et al. (2014) was developed considering a given locus that segregates in $1:1:1:1$ fashion. This would result in four different conditional probabilities for QTL genotypes ($P^1Q^1$, $P^1Q^2$, $P^2Q^1$, $P^2Q^2$). However, for situations in which the genetic map is composed with partially informative markers may be limitations on the level of information that the odds may contain. An such example is a genome region where the those four genotypic classes are not present.

The statistical-genetic model of Gazaffi et al. (2014) was developed considering a locus with the four genotypic classes. With such classes, it is possible to model until three genetic effects using three contrasts: additive effects of the QTL alleles of parent P, additive effects of the QTL alleles of parent Q, and the intra-locus interaction (dominance) between both additive effects.  More information at the Methodology section of Gazaffi et al. (2014). However, depending upon the markers presenting in the study and dispite the use of multipoint probabilities, some regions of the linkage map could no have enough information to estimate all the genetic effects. This may occur in cases where a linkage group has only one or two marker types (*e.g.*, $C$ and $D_1$, or, $C$ and $D_2$) more information about marker types at Wu et al. (2002a).

In order to identify colinearity problems between the QTL conditional probabilities, we used a criteria described in Belsley et al. (1980) based on diagnosis of singular value decomposition using condition index. Belsley et al. (1980) suggests collinearity problems occur at condition index over 30. However, we had preliminary results in QTL mapping where a threshold of 3.5 provided good results.

In cases where the problems of colinearity do not verify, it is used the contrasts 1, 2, and 3 of the contrasts matrix to obtain the mapping model. Seggregation and linkage phase are inferred using the proceeding presented by Gazaffi et al. (2014). Although, when it is identified a dependence relationship between QTL genotypes probabilities, it is suggested the removal ou adaptation of the contrasts to define the genetic effects (contrasts 4 to 13). Table 1 estabilishes all the contrasts that the algorithm can verify. The notation in this section is the same of Gazaffi et. al (2014). In this same paper is described the contrasts 1, 2, and 3. Contrasts 1 and 2 are similar to a backcross model. Contrasts 3 and 9 are similar to an $F_2$ model.

<center>
**Table 1: Contrasts**


| Contrast |&nbsp; &nbsp; &nbsp; &nbsp;| $P^{1}Q^{1}$ | $P^{1}Q^{2}$ | $P^{2}Q^{1}$ | $P^{2}Q^{2}$ |
|:--------:|:----:|:------------:|:------------:|:------------:|:------------:|
| 1        |&nbsp; &nbsp;| 1            | 1            | -1           | -1           |
| 2        |&nbsp;| 1            | -1           | 1            | -1           |
| 3        |&nbsp;| 1            | -1           | -1           | 1            |
| 4        |&nbsp;| 0            | 0            | 1            | -1           |
| 5        |&nbsp;| 1            | -1           | 0            | 0            |
| 6        |&nbsp;| 0            | 1            | 0            | -1           |
| 7        |&nbsp;| 1            | 0            | -1           | 0            |
| 8        |&nbsp;| 0            | 1            | -1           | 0            |
| 9        |&nbsp;| 1            | 0            | 0            | -1           |
| 10       |&nbsp;| 1/3          | 1/3          | 1/3          | -1           |
| 11       |&nbsp;| 1/3          | 1/3          | -1           | 1/3          |
| 12       |&nbsp;| 1/3          | -1           | 1/3          | 1/3          |
| 13       |&nbsp;| -1           | 1/3          | 1/3          | 1/3          |

</center>

When it is used the original contrasts (1, 2, 3) to perform the QTL mapping, the segregation pattern is inferred based on the following complementary hypotheses. First, it is verified the significance of the effects $\alpha_p^{\ast}$, $\alpha_q^{\ast}$ e $\delta_{pq}^{\ast}$ using the hypotheses: $H_{01}: \alpha_p^{\ast}=0$, $H_{02}: \alpha_q^{\ast}=0$, and $H_{03}: \delta_{pq}^{\ast}=0$, respectivally. If it is observed more than one significant effect, it is checked the following additional hypotheses to test the equality of the estimates: $H_{04}:$ $|\alpha_p^{\ast}|=|\alpha_q^{\ast}|$, $H_{05}:$ $|\alpha_p^{\ast}|$ = $|\delta_{pq}^{\ast}|$, and $H_{06}:$ $|\alpha_q^{\ast}|=|\delta_{pq}^{\ast}|$. If just one genetic effect is significant, it is assumed the QTL has segregation 1:1. If two genetic effects are significant, a complementary test is performed to infer if the segregation pattern is $1:2:1$ (non-rejected complementary hypothesis) or $1:1:1:1$ (rejected complementary hypothesis). If there are three significant genetic effects, it is considered the QTL is: i) $3:1$ if the three complementary hypotheses were not rejected, ii) $1:2:1$ if just one between the three hypotheses was not rejected, and, iii) $1:1:1:1$ for other cases. The linkage phase between QTL and markers is inferred through the signal interpretation of $\alpha_p^{\ast}$ and $\alpha_q^{\ast}$. The default model assumes the configuration:

<center>
$P^1_mP^1P^1_{m+1}/P^2_mP^2P^2_{m+1} \times Q^1_mQ^1Q^1_{m+1}/Q^2_mQ^2Q^2_{m+1}$
</center>

Which is the both parents on coupling phase. If this is the case, the estimates of  $\alpha^{\ast}_p$ and $\alpha^{\ast}_q$ will be positives. In the other configurations, the estimates will show inverted signals (Gazaffi et al 2014).

On the other hand, it is not possible to use a model  with all the parameters in a less informative genome region. In such cases, the QTLs characterization are restrained. Within Table 2 (based upon the dependence relationship between QTLs genotypes), it is possible to infer which QTL effects are estimable, as well as their expectations and characterizations.

<center>
**Table 2: Identification of estimable effects and QTL characterization from the possible colinearity relationships between genotypes and mapped QTL$^{*}$**

<font size="0.5">

| Case |&nbsp; &nbsp; &nbsp; &nbsp;| Colinearity			|&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;| *D* contrasts	|&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;| Estimable effects|&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;| Contrast espected mean                      		|&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;| Rejected hypothesis     |&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;| Observed segregation$^{**}$ |&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;|         Phases |
|:----:|:----:|:-------------------------------------:|:----:|:--------------------:|:----:|:--------------------:|:----:|:------------------------------------------------------------:|:----:|:-----------------------------:|:----:|:--------------------------:|:----:|:--------------------:|
|   i  |&nbsp; &nbsp; &nbsp; &nbsp;|$P^1Q^1 = P^1Q^2$ and $P^2Q^1 = P^2Q^2$|&nbsp; &nbsp; &nbsp; &nbsp;| $(1)$                |&nbsp; &nbsp; &nbsp; &nbsp;| $\alpha_p^\ast$      |&nbsp; &nbsp; &nbsp; &nbsp;| $\alpha_p^\ast$						|&nbsp; &nbsp; &nbsp; &nbsp;|$H_{01}$: $\alpha_p^{\ast} = 0$|&nbsp; &nbsp; &nbsp; &nbsp;| 1:1                        |&nbsp; &nbsp; &nbsp; &nbsp;| $P^1P^2$ or $P^2P^1$ |
|  ii  ||$P^1Q^1 = P^2Q^1$ and $P^1Q^2 = P^2Q^2$|| $(2)$                || $\alpha_q^\ast$      || $\alpha_q^\ast$                                              ||$H_{02}$: $\alpha_q^{\ast}=0$  || 1:1                        || $Q^1Q^2$ or $Q^2Q^1$ |
|  iii ||$P^1Q^1 = P^1Q^2$                      || $(1)$                || $\alpha_p^{\ast}$    || $\alpha_p^{\ast}$                                            ||$H_{01}$                       || 1:1                        || $P^1P^2$ or $P^2P^1$ |
|      ||                                       || $(4)$                || $\alpha_q^{\ast}$    || $(1/2)(\alpha_q^{\ast}-\delta_{pq}^{\ast})$                  ||$H_{02}$                       || 1:2:1                      ||   unsolved           |
|      ||                                       ||                      ||                      ||                                                              ||$H_{01}$ and  $H_{02}$         || 1:2:1 or 3:1               || $P^1P^2$ or $P^2P^1$ |
|  iv  ||$P^2Q^1 = P^2Q^2$                      || $(1)$                || $\alpha_p^{\ast}$    || $\alpha_p^{\ast}$                                            ||$H_{01}$                       || 1:1                        || $P^1P^2$ or $P^2P^1$ |
|      ||                                       || $(5)$                || $\alpha_q^{\ast}$    || $(1/2)(\alpha_q^{\ast}+\delta_{pq}^{\ast})$                  ||$H_{02}$                       || 1:2:1                      ||   unsolved           |
|      ||                                       ||                      ||                      ||                                                              ||$H_{01}$ and $H_{02}$          || 1:2:1 or 3:1               || $P^1P^2$ or $P^2P^1$ |
|   v  ||$P^1Q^1 = P^2Q^1$                      || $(6)$                || $\alpha_p^{\ast}$    || $(1/2)(\alpha_p^{\ast}-\delta_{pq}^{\ast})$                  ||$H_{01}$                       || 1:2:1                      ||   unsolved           |
|      ||                                       || $(2)$                || $\alpha_q^{\ast}$    || $\alpha_q^\ast$                                              ||$H_{02}$                       || 1:1          	       || $Q^1Q^2$ or $Q^2Q^1$ |
|      ||				      	|| 		      	||			|| 						                ||$H_{01}$ and  $H_{02}$  	 || 1:2:1 or 3:1	       || $Q^1Q^2$ or $Q^2Q^1$ |
|  vi  ||$P^1Q^2 = P^2Q^2$		       	|| $(7)$		|| $\alpha_p^{\ast}$    || $(1/2)(\alpha_p^{\ast}+\delta_{pq}^{\ast})$          	||$H_{01}$      	 	 || 1:2:1    	               ||   unsolved   	       |
|      ||                 		       	|| $(2)$ 	      	|| $\alpha_q^{\ast}$    ||  $\alpha_q^\ast$                                             ||$H_{02}$	                 || 1:1			       || $Q^1Q^2$ or $Q^2Q^1$ |
|      ||                   		       	||         	      	||                      ||                                                              ||$H_{01}$ and  $H_{02}$	 || 1:2:1 or 3:1 	       || $Q^1Q^2$ or $Q^2Q^1$ |
|  vii ||$P^1Q^1 = P^2Q^2$		       	|| $(8)$ 	      	|| $\alpha_p^{\ast}$    ||  $(1/2)(\alpha_p^{\ast}-\alpha_{q}^{\ast})$                  ||$H_{01}$                	 || 1:2:1       	       ||   unsolved  	       |
|      || 	  			       	|| $(3)$	      	|| $\delta_{pq}^{\ast}$ || $\delta_{pq}^\ast$   			   		||$H_{03}: \delta_{pq}^{\ast}=0$ || 1:1			       ||   unsolved	       |
|      ||      				       	||                      ||            	        ||  				 	   		    	||$H_{01}$ and  $H_{03}$ 	 || 1:2:1 or 3:1	       ||   unsolved           |
| viii ||$P^1Q^2 = P^2Q^1$		       	|| $(9)$ 	      	|| $\alpha_p^{\ast}$    || $(1/2)(\alpha_p^{\ast}+\alpha_{q}^{\ast})$  		    	||$H_{01}$                	 || 1:2:1         	       ||   unsolved           |
|      || 		  		       	|| $(3)$		|| $\delta_{pq}^{\ast}$ || $\delta_{pq}^\ast$                          		    	||$H_{03}$                	 || 1:1          	       ||   unsolved	       |
|      ||         			       	||                      ||                      ||  					   		    	||$H_{01}$ and  $H_{03}$	 || 1:2:1 or 3:1	       ||   unsolved	       |
|  ix  ||$P^1Q^1 = P^1Q^2 = P^2Q^1$	       	|| $(10)$ 	      	|| $\alpha_p^{\ast}$    || $(4/3)( \alpha_p^{\ast}+\alpha_p^{\ast}-\delta_{pq}^{\ast})$ ||$H_{01}$ 	                 || 3:1      		       ||   unsolved	       |
|  x   ||$P^1Q^1 = P^1Q^2 = P^2Q^2$	       	|| $(11)$ 	      	|| $\alpha_p^{\ast}$    || $(4/3)( \alpha_p^{\ast}-\alpha_p^{\ast}+\delta_{pq}^{\ast})$ ||$H_{01}$                       || 3:1      		       ||   unsolved	       |
|  xi  ||$P^1Q^1 = P^2Q^1 = P^2Q^2$	       	|| $(12)$ 	      	|| $\alpha_p^{\ast}$    || $(4/3)(-\alpha_p^{\ast}+\alpha_p^{\ast}+\delta_{pq}^{\ast})$ ||$H_{01}$               	 || 3:1 		       ||   unsolved	       |
|  xii ||$P^1Q^2 = P^2Q^1 = P^2Q^2$	       	|| $(13)$ 	      	|| $\alpha_p^{\ast}$    || $(4/3)(-\alpha_p^{\ast}-\alpha_p^{\ast}-\delta_{pq}^{\ast})$ ||$H_{01}$               	 || 3:1 		       ||   unsolved	       |

</center>

($*$) For clarity reasons, only the QTL alleles are represented; $P^1P^2$ represents the configuration $P^1_mP^1P^1_{m+1}/P^2_mP^2P^2_{m+1}$, and so on. If it is considered a cross between two parents $P$ and $Q$ genetically equal (or an inbreeding phenomenon), the real segregation would be equivalent to the observed in the mapping population ($1:2:1$) with known linkage phases between QTLs and markers as $P^1P^2 \times Q^1Q^2$ or $P^2P^1 \times Q^2Q^1$.
($**$): Due to colinearity between QTLs genotypes probabilities it is not possible to estimate the real QTL segregation. As some of the effects are not possible to independently estimate, the QTL can be just partialy characterized using the available information.

</font>

For cases $i$ and $ii$, it only possible to estimate one parent effect, which allows to detect a QTL with 1:1 pattern (similar to a backcross population). Yet, some considerations about it: 

- For a QTL presented in a region with such colinearity, it is only possible to test the QTL presence segregating in one parent. In this case, even if the QTL would present more significant effects they would not have a possible estimation. This is due to low marker information which indicates that the observed QTL segregation would only be partially consistent with the real QTL segregation.

- Even though some regions permit QTL detection with a genetic effect similar to a backcross - which can be compared to the fullsib progeny of a pseudo-testcross approach (Grattapaglia and Sederoff, 1994) - the present model has as advantage the possiblity to incorporate cofactors, regions with different segregation pattern with respect to the mapping local. In this way, it increases the statistical power to the QTL detection.

In the case $iii$, it is possible to estimate the effect of  $\alpha_p^{\ast}$. However, due to the conditional probabilities of genotypes $P^1Q^1$ and $P^1Q^2$ being the same, it is not possible to isolate the effects $\alpha_q^\ast$ e $\delta_{pq}^\ast$. In such case, the strategy of the algorithm is to test the existence of a genetic effect $\alpha_q^\ast$ within the informative genotypes ($P^2Q^1$ and $P^2Q^2$) - contrast 4 of Table 1. However, this contrast tests a possible combination between $\alpha_q^\ast$ and $\delta_{pq}^\ast$ effects which would difficult identification of the effect origin and its correct segregation pattern. Moreover, if only the effect $\alpha_p^{\ast}$ is significant, the QTL would segregate in this progeny in a $1:1$ fashion. If only the effect  $\alpha_q^\ast$ is significant, the segregation pattern in the progeny would be in the $1:2:1$ proportion. In cases where both effects are significants, the observed segregation will be $1:2:1$ or $3:1$. The $3:1$ pattern rises when the absolute value of $\alpha_q^\ast$ is the double of the absolute value of $\alpha_p^\ast$. In this type of colinearity the identification of the QTL segregation pattern - as well as the linkage phase between the markers - is partially informative because there are among markers which are not possible to distinctly estimate their genetic effects. Therefore, the future input of markers with different segregation patterns could provide results more precises for the QTL mapping, even changing the segregation pattern ealier inferred. This same situation can be analysed for cases $iv$, $v$, $vi$, $vii$, and $viii$.

In the case $viii$, if it is assumed an self-polinization or a cross between two parents genetically equals, it can be assumed a situation similar to a $F_2$ population, therefore,  $\alpha_p^\ast$ and $\alpha_q^\ast$ would be equal providing a unbiased estimate for the segregation and linkage phase between QTLs and markers $(P^1P^2 \times Q^1Q^2$ (if $\alpha_p^\ast$ positive) or $P^2P^1 \times Q^2Q^1$ (if $\alpha_p$ negative). To obtain the dominance estimate as proposed by Cockerham (1954), it should be considered half of the effect of $\delta_{pq}^\ast$.

For cases from $ix$ to $xii$, the colinearity is originated due to the similarity between three genotypic classes which only allows to estimate a genetic effect represented by a linear combination between $\alpha_p^{\ast}$, $\alpha_q^{\ast}$ and $\delta_{pq}^{\ast}$. In such cases, it is not possible the correct understanding of the segregation and linkage phases between QTLs and markers. Moreover, it is emphasized that to limit those contrasts it was used the coefficient $1/3$ for similar genotypes and $-1$ for the different genotype. Such contrasts are not orthogonals in relation to the other and can leads to a burdensome QTL mapping process. As an example, using Multiple Interval Mapping models to epistasis study. Thus, other strategies may be applied when this type of situation is verified.

<a name="appendixb">
## Appendix B: Graphical options with plot_fullsibQTL
</a>
In this section we show how to plot different graphics with the `plot_fullsibQTL` function. A complete description can be viewed through `?plot_fullsibQTL`. As an example we will use the example data from the package:

```{r}
  library(fullsibQTL)
  data("example_QTLfullsib")
  fullsib <- create_fullsib(example_QTLfullsib,
                             list(LG1_final, LG2_final, LG3_final, LG4_final),
                             step=1,map.function="kosambi",condIndex=3.5)

  ## IM
  im <- im_scan(fullsib)
  qtls.im <- r2_ls(fullsib, pheno.col=1, lg=c(1,1,2,2,2,3,3,4),
                 pos=c("loc13","M12","loc45","loc77","loc108","M33","loc65","M52"))
  ## CIM
  cofs.fs <- cof_selection(fullsib, pheno.col=1, k=log(300), n.cofactor = 10)
  cim<-cim_scan(cofs.fs)
  qtls.cim <- r2_ls(fullsib, pheno.col=1, lg=c(1,1,2,2,2,3,3,4),
                    pos=c("loc13","M12","loc45","loc77","loc108","M33","loc65","M52"))
```
### Customizing Graphics

The minimal plot has to have the fullsib object and at list one result of `fullsib_scan`. At the bottom of the graphic it is shown the genetic map.
```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=cim)
```

We can set the legend of the QTL mapping with the `qtlmapping` argument:
```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=cim,
                qtlmapping="CIM")
```

### Graphical comparision between different `(c)im_scan`
There is the option to insert multiple results from `(c)im_scan` with listing them with `list()`. Those multiple results can be different traits, a same trait in different environments, comparision of IM $\times$ CIM, and so on. Here we compare IM $\times$ CIM

```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                qtlmapping=c("CIM","IM"))
```

### Adding QTL location to the graphic
You can insert the QTLs position from the `r2_ls` function for each one of the `(c)im_scan`. In order to do it, you need to use the argument `r2ls.out`:

```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=cim,
                r2ls.out=qtls.cim,
                qtlmapping="CIM")
```

For multiple QTLs mappings:

```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"))
```

To set threshold lines, based on the `summary(cim_perm)` peaks 1 and 2 with $\alpha=0.05) results:

```{r}
summary(cim_perm)

plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906))
```

For printing purposes, another option is to change the graphic for a grayscale:

```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906),
                grayscale=TRUE)
```

There is the option to set which linkage groups you want to plot. To plot just LG 2:

```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906),
                lgs=2)
```

To plot LG 2 and 4:
```{r}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906),
                lgs=c(2,4))
```

### Interactive plot
There is also the option to plot a interactive graphic. Inside the function, we use the package `plotly` as the engine. With the option `interact=TRUE` it will be created an interactive `html` graphic in you working directory folder or you can set the folder name to create the `html`. The plot will be returned in your default browser. If you do not want to open the browser, set `browser=FALSE`

```{r, eval=FALSE}
## Setting the file name
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906), 
                interact=TRUE, 
                file="cim_im.html")

## Setting the folder name
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906), 
                interact=TRUE, 
                file="cim_im.html",
                folder="/home/your_user_name/Pictures")

## Setting to not open the browser
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906), 
                interact=TRUE, 
                file="cim_im.html",
                folder="/home/your_user_name/Pictures",
                browser=FALSE)
```

The above script creates the following graphic. It is a interactive `plotly`. *Click on legend entries to toggle traces, click-and-drag on the chart to zoom, double-click to autoscale, shift-and-drag to pan* ([plotly website](https://plot.ly/r/getting-started/)). Additionaly, it is included in the tooltip the LOD score, the QTL estimated effect, and marker names in the map.

<iframe src='http://statgen.esalq.usp.br/fullsibQTL_graphics/cim_im_greens/cim_im.html', height="600", width = "800">

</iframe>

The user can also set the height and width of the interact plot within the arguments `height` and `width`, they are the same as the arguments of `plotly` function. The default of `height` is 500 pixels and for `width` is 250 $\times$ the number of LGs to be plot. All the arguments can be combined in the interactive plot. 

### Changing colors

There is the option to choose the colors for each LG, the argument `lg.colors` will pass its values to the `ggplot2::scale_colour_manual` function. You can check more colors options at this [cheatsheet](https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/colorPaletteCheatsheet.pdf) or at this color [guide](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf).

Here we plot an interactive plot in order to compare the CIM and IM mapping setting two threshold values, `darkgreen` color for CIM and `gree` for IM, and choosing to plot just LG 2 and 3.

```{r, eval=FALSE}
plot_fullsibQTL(fullsib=fullsib,
                fullsib.scan=list(cim,im),
                r2ls.out=list(qtls.cim,qtls.im),
                qtlmapping=c("CIM","IM"),
                thr = c(4.023724, 2.900906), 
                lgs = c(2,4),
                interact=TRUE, 
                lg.colors=c("darkgreen","green"),
                height=600,
                width=800,
                file="cim_im_greens.html",
                folder="/home/your_user_name/Pictures")
```

<iframe src='http://statgen.esalq.usp.br/fullsibQTL_graphics/cim_im_greens.html', height="650", width = "800">

</iframe>

<a name="appendixc">
## Appendix C: Graphical options with plot
</a>
In this section we show how to plot different graphics with the `plot` function. A complete description can be viewed through `?plot.fullsib_scan`. As an example we will use the example data from the package:

```{r}
  library(fullsibQTL)
  data("example_QTLfullsib")
  fullsib <- create_fullsib(example_QTLfullsib,
                             list(LG1_final, LG2_final, LG3_final, LG4_final),
                             step=1,map.function="kosambi",condIndex=3.5)

  ## IM
  im <- im_scan(fullsib)
  qtls.im <- r2_ls(fullsib, pheno.col=1, lg=c(1,1,2,2,2,3,3,4),
                 pos=c("loc13","M12","loc45","loc77","loc108","M33","loc65","M52"))
  ## CIM
  cofs.fs <- cof_selection(fullsib, pheno.col=1, k=log(300), n.cofactor = 10)
  cim<-cim_scan(cofs.fs)
  qtls.cim <- r2_ls(fullsib, pheno.col=1, lg=c(1,1,2,2,2,3,3,4),
                    pos=c("loc13","M12","loc45","loc77","loc108","M33","loc65","M52"))

```

The basic graphic


You can check more colors options at this [cheatsheet](https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/colorPaletteCheatsheet.pdf) or at this color [guide](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf) 

### Customizing Graphics
```{r}
plot(im)

## argument lty
## Changes the line type
## There is a total of 6 different line types as follow:
plot(im)
abline(h=c(1,2,3,4,5,6),lty=c(1,2,3,4,5,6))

## argument lwd
## Changes the line width
## It is changed in a relative proportion to the default. 2 is twice the default width, 3 is triple of the default, and so on.
plot(im)
abline(h=c(.5,1,2,3,4,5,6),lwd=c(.5,1,2,3,4,5,6))

## You can set any of them with the plot with:
plot(im, lty=3, lwd=2)

## argument lg
## Plotting just LG1
plot(im, lty=3, lwd=2, lg=1)

## Plotting LG1 and LG2
plot(im, lty=3, lwd=2, lg=c(1,2))

## argument label.lg
## Labels the LGs
plot(im, lty=3, lwd=2, lg=c(1,2), label.lg=c("Chr 1","Chr 2"))

## argument cex.axis  
## Changes the magnification of axis annotation relative to cex
plot(im, lty=3, lwd=2, lg=c(1,2), label.lg=c("Chr 1","Chr 2"), cex.axis=1)
plot(im, lty=3, lwd=2, lg=c(1,2), label.lg=c("Chr 1","Chr 2"), cex.axis=1.5)

## argument incl.mkr
## Includes the marker position
## There are 3 options: `points`, `name`, and, `none`. `none` is default.
plot(im, lty=3, lwd=2, lg=c(1,2), label.lg=c("Chr 1","Chr 2"), cex.axis=1.2, incl.mkr="points")

## argument cex.incl
## Changes the magnification of incl.mrk annotation relative to cex.
plot(im, lty=3, lwd=2, lg=c(1,2), label.lg=c("Chr 1","Chr 2"), cex.axis=1.2, incl.mkr="name", cex.incl = .75)

## argument gap
## Enlarges the gap between chromosomes (in CM)
plot(im, lty=3, lwd=2, lg=c(1,2), label.lg=c("Chr 1","Chr 2"), cex.axis=1.2, incl.mkr="name", cex.incl = .75, gap = 50)

## argument col
## Changes the line colors
plot(im, lty=3, lwd=2, lg=c(1,2), label.lg=c("Chr 1","Chr 2"), cex.axis=1.2, incl.mkr="name", cex.incl = .75, gap = 50, col="red")
```

You can check more colors options at this [cheatsheet](https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/colorPaletteCheatsheet.pdf) or at this color [guide](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf) 

You can also customize the plot with the default R plot options as `main`, `ylab`, `xlab`, `yaxt`, `ylim`, etc. 

```{r}
plot(im, lty=1, lwd=2, lg=c(1,2), cex.axis=1.2, incl.mkr="name", cex.incl = .9, gap = 50, col="red", alternate.lgid = TRUE,
     ## Configuring with R plot options:
     main = "IM - Trait - Crop", 
     xlab = "Linkage Group", 
     ylab = "LOD score", 
     yaxt = "n",
     ylim = c(0,10))
axis(side=2,at=seq(0,10,1),lwd=1)
abline(h=im.peak1, lty=2)
abline(h=im.peak2, lty=3)
legend("topleft", lty=c(1,2,3,0), cex=0.75, col=c("red","black","black"), legend=c("IM","1st Peak Threshold","2nd Peak Threshold"))
```

### Graphical comparision between `(c)im_scan`
The IM and CIM mapping results can be compared in a graphical way with the following codes:
```{r}
plot(im,cim, lty = c(2,1), lwd = 2, incl.mkr = "name", cex.incl = 0.7, gap=50,
     cex.axis = 0.8, col = c("blue","red"),
     main = "IM - Trait - Crop", 
     xlab = "Linkage Group", 
     ylab = "LOD score", 
     yaxt = "n",
     ylim = c(0,15))
axis(side=2,at=seq(0,15,1),lwd=1)
abline(h = im.peak1, lty = 2, lwd = 0.75)
abline(h = cim.peak1, lty = 2, lwd = 0.75)
legend("topleft", lty=c(2,1,2,2), cex=0.5, col=c("red","blue","black","black"), legend=c("IM","CIM","1st Threshold","2nd Threshold"))


## Adding QTLs
```
### Adding QTL location to the graphic
Our QTLs are the following:
```{r}
print(cim,lg=1,pos=c("loc13","M12"))
print(cim,lg=2,pos=c("loc45","loc77","loc108"))
print(cim,lg=3,pos=c("M33","loc65"))
print(cim,lg=4,pos="M52")
```
To add them to the graphic we can use the function `points` as follows. Knowing that the `gap` space is `50`, we can add this to the total size of each LG and get the position. The points will be localized at:
```{r}
points.loc <- c(13,110,45,77,108,20,65,60)
points.loc <- points.loc + c(0,0,
                             140+50,140+50,140+50,
                             140+50+140+50,140+50+140+50,
                             140+50+140+50+140+50)
```
Therefore:
```{r}
plot(im,cim, lty = c(2,1), lwd = 2, incl.mkr = "name", cex.incl = 0.5, gap=50,
     cex.axis = 0.8, col = c("blue","red"),
     main = "IM - Trait - Crop", 
     xlab = "Linkage Group", 
     ylab = "LOD score", 
     yaxt = "n",
     ylim = c(-2,15))
axis(side=2,at=seq(0,15,1),lwd=1)
abline(h = im.peak1, lty = 2, lwd = 0.75)
abline(h = cim.peak1, lty = 2, lwd = 0.75)
points(y=rep(0,8),x=points.loc, pch=17, col="dark green")
legend("topleft", lty=c(2,1,2,2,0), cex=0.75, pch = c(NA,NA,NA,NA,17),col = c("red", "blue", "black", "black", "dark green"), legend = c("IM","CIM","1st Threshold","2nd Threshold","QTL"),bty='n',y.intersp=.75)
```

## References
Belsley, D. A., Edwin K., and Roy E. W. (2005). **_Regression diagnostics: Identifying influential data and sources of collinearity_**. Vol. 571. John Wiley | Sons.

Cockerham, C. C. (1954). An extension of the concept of partitioning hereditary variance for analysis of covariances among relatives when epistasis is present. **_Genetics_** 39(6), 859.

Chen, L., and Storey, J. D. (2006). Relaxed significance criteria for linkage analysis. **_Genetics_**, 173(4), 2371-2381.

Churchill, G. A., and Doerge, R. W. (1994). Empirical threshold values for quantitative trait mapping. **_Genetics_**, 138(3), 963-971.

Gazaffi, R., Margarido, G. R., Pastina, M. M., Mollinari, M., Garcia, A. A. F. (2014). A model for quantitative trait loci mapping, linkage phase, and segregation pattern estimation for a full-sib progeny. **_Tree Genetics & Genomes_**, 10(4), 791-801.

Grattapaglia, D., and Sederoff, R. (1994). Genetic linkage maps of Eucalyptus grandis and Eucalyptus urophylla using a pseudo-testcross: mapping strategy and RAPD markers. **_Genetics_**, 137(4), 1121-1137.

Margarido, G. R. A., Souza, A.P. and Garcia, A. A. F. OneMap: software
for genetic mapping in outcrossing species. **_Hereditas_** 144,
78-79, 2007.

Wu, R., Ma, C.X., Painter, I. and Zeng, Z.-B. Simultaneous maximum
likelihood estimation of linkage and linkage phases in outcrossing
species. **_Theoretical Population Biology_** 61, 349-363, 2002a.

Wu, R., Ma, C.-X., Wu, S. S. and Zeng, Z.-B. Linkage mapping of
sex-specific differences. **_Genetical Research_** 79, 85-96, 2002b.

```{r, eval=FALSE,echo=FALSE}

## Onemap References
# 
# Adler, J. **_R in a Nutshell_** A Desktop Quick Reference, 2009.
# 
# Broman, K. W., Wu, H., Churchill, G., Sen, S., Yandell, B. **_qtl:
# Tools for analyzing QTL experiments_** R package version
# 1.09-43, 2008. [http://www.rqtl.org/](http://www.rqtl.org/)
# 
# Buetow, K. H., Chakravarti, A. Multipoint gene mapping using
# seriation. I. General methods. **_American Journal of Human
# Genetics_** 41, 180-188, 1987.
# 
# Doerge, R.W. Constructing genetic maps by rapid chain delineation.
# **_Journal of Agricultural Genomics_** 2, 1996.
# 
# Garcia, A.A.F., Kido, E.A., Meza, A.N., Souza, H.M.B., Pinto, L.R.,
# Pastina, M.M., Leite, C.S., Silva, J.A.G., Ulian, E.C., Figueira, A.
# and Souza, A.P. Development of an integrated genetic map of a
# sugarcane _Saccharum spp._ commercial cross, based on a
# maximum-likelihood approach for estimation of linkage and linkage
# phases. **_Theoretical and Applied Genetics_** 112, 298-314, 2006.
# 
# Haldane, J. B. S. The combination of linkage values and the
# calculation of distance between the loci of linked factors. **_Journal
# of Genetics_** 8, 299-309, 1919.
# 
# Jiang, C. and Zeng, Z.-B. Mapping quantitative trait loci with
# dominant and missing markers in various crosses from two inbred lines.
# **_Genetica_** 101, 47-58, 1997.
# 
# Kosambi, D. D. The estimation of map distance from recombination
# values. **_Annuaire of Eugenetics_** 12, 172-175, 1944.
# 
# Lander, E. S. and Green, P. Construction of multilocus genetic linkage
# maps in humans. **_Proc. Natl. Acad. Sci. USA_** 84, 2363-2367, 1987.
# 
# Lander, E.S., Green, P., Abrahanson, J., Barlow, A., Daly, M.J.,
# Lincoln, S.E. and Newburg, L. MAPMAKER, An interactive computing
# package for constructing primary genetic linkage maps of experimental
# and natural populations. **_Genomics_** 1, 174-181, 1987.
# 
# Lincoln, S. E., Daly, M. J. and Lander, E. S. Constructing genetic
# linkage maps with MAPMAKER/EXP Version 3.0: a tutorial and reference
# manual. **_A Whitehead Institute for Biomedical Research Technical
# Report_** 1993.
# 
# Matloff, N. **_The Art of R Programming_**. 2011. 1st ed. San
# Francisco, CA: No Starch Press, Inc., 404 pages.
# 
# 
# 
# Mollinari, M., Margarido, G. R. A., Vencovsky, R. and Garcia, A. A. F.
# Evaluation of algorithms used to order markers on genetics maps.
# **_Heredity_** 103, 494-502, 2009.
# 
# Oliveira, K.M., Pinto, L.R., Marconi, T.G., Margarido, G.R.A.,
# Pastina, M.M., Teixeira, L.H.M., Figueira, A.M., Ulian, E.C., Garcia,
# A.A.F., Souza, A.P. Functional genetic linkage map on EST-markers for
# a sugarcane (_Saccharum spp._) commercial cross. **_Molecular
# Breeding_** 20, 189-208, 2007.
# 
# Oliveira, E. J., Vieira, M. L. C., Garcia, A. A. F., Munhoz, C.
# F.,Margarido, G. R.A., Consoli, L., Matta, F. P., Moraes, M. C.,
# Zucchi, M. I., and Fungaro,M. H. P. An Integrated Molecular Map of
# Yellow Passion Fruit Based on Simultaneous Maximum-likelihood
# Estimation of Linkage and Linkage Phases **_J. Amer. Soc. Hort.
# Sci._** 133, 35-41, 2008.
# 
# Tan, Y., Fu, Y. A novel method for estimating linkage maps.
# **_Genetics_** 173, 2383-2390, 2006.
# 
# Van Os H, Stam P, Visser R.G.F., Van Eck H.J. RECORD: a novel method
# for ordering loci on a genetic linkage map. **_Theor Appl Genet_**
# 112, 30-40, 2005.
# 
# Voorrips, R.E. MapChart: software for the graphical presentation of
# linkage maps and QTLs. **_Journal of Heredity_** 93, 77-78, 2002.
# 
# Wang S., Basten, C. J. and Zeng Z.-B. Windows QTL Cartographer 2.5.
# Department of Statistics, North Carolina State University, Raleigh,
# NC, 2010.
# [http://statgen.ncsu.edu/qtlcart/WQTLCart.htm](http://statgen.ncsu.edu/qtlcart/WQTLCart.htm)
# 
# Wu, R., Ma, C.X., Painter, I. and Zeng, Z.-B. Simultaneous maximum
# likelihood estimation of linkage and linkage phases in outcrossing
# species. **_Theoretical Population Biology_** 61, 349-363, 2002a.
# 
# Wu, R., Ma, C.-X., Wu, S. S. and Zeng, Z.-B. Linkage mapping of
# sex-specific differences. **_Genetical Research_** 79, 85-96, 2002b.
```




